### PR TITLE
Task/4175 export diary entries

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds the export functionality to the store and connects it to the UI.

## Link to Jira
[4175](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4175)

## Screenshots
RPReplay_Final1608116348

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

**Review by KaiTeuber — APPROVED** [review]

**Comment by ouboub** [comment]
Not sure I abuse again the request but I still think an important feature is missing here.
the video only shows how to export all entries, which is fine for the Gesundheitsamt, but since they are overwhelmed with work are hampered by software incompatibilities. 
That is why it would be very important to inform my contacts myself, but because of privacy reasons, I don't want to inform all my contacts about all my meetings. I don't understand that the APP is so picky with respect to data protection in certain aspects but in this aspect seems not to care at all.
Did I explain my point?

**Comment by christianneu** [comment]
@ouboub: I understand your point, but as I already pointed out last time, please refer your improvements in your wishlist entry as the community is then able to discuss it there with you. 

The fact that you posted it in the pull request does not help as it is going down during development and in a normal lifecycle every pull request will be merged or closed and deleted anyway.

Best Regards
Christian 
Corona Warn-App Open Source Team

**Comment by ouboub** [comment]
@christianneu 
Well I am more acquainted with discussion of patches in mailing lists. In these cases context one would not only discuss syntactic issues of the patch (or pull request), but also the question whether implementing Feature A would make it more difficult later to implement Feature B, which includes Feature A

**Comment by GisoSchroederSAP** [comment]
Dear @ouboub,

I'd like to emphasize three points here:

1. As my colleagues already did, we have to insist that all improvement/enhancement request and discussions for an existing feature a future functions are documented in the cwa-wishlist repository.
2. The new function of the contact diary is an early version and we are well aware there is room for improvement.
3. And no, you are wrong with the comment mentioned:

> in certain aspects but in this aspect seems not to care at all.

Data privacy was and still is an omnipresent topic during design, implementation and test of the CWA functions. You might want to check the existing blog entry (https://www.coronawarn.app/de/blog/2020-12-28-corona-warn-app-version-1-10/) for the intention of the contact diary feature. So, if your intention is not to share certain contacts with the health authority, this is your responsibility. But obviously, there is no need to share your personal list of documented meetings with all of you contacts - again, this is your data and you select carefully what you share and who receives these information.

Side note: I'll copy this response to the issue (https://github.com/corona-warn-app/cwa-app-ios/pull/1651):
Best,
Giso
Corona Warn-App Open Source Team

**Comment by ouboub** [comment]
>>> "G" == GisoSchroederSAP  <notifications@github.com> writes:
Dear @GisoSchroder


> Dear @ouboub, 
> I'd like to emphasize three points here:

I am not entirely sure where to answer this, without getting pointed out
that I abuse the pull requests. (Sorry)

> 1. As my colleagues already did, we have to insist that all
> improvement/enhancement request and discussions for an existing
> feature a future functions are documented in the
> [cwa-wishlist](https://github.com/corona-warn-app/cwa-wishlist)
> repository.

Ok fair enough. Do you want me to create a new issue there for example, 

feedback and enhancement of the new contact diary functionality 

> 2. The new functio of the contact diary is an early version and we are
> well aware there is room for improvement.

Yes, sure.

> 3. And no, you are wrong with the comment mentioned in the referred
> issue (https://github.com/corona-warn-app/cwa-app-ios/pull/1695) that
> was incorrectly placed as pull-request:

>> in certain aspects but in this aspect seems not to care at all.

> Data privacy was and still is an omnipresent topic during design,
> implementation and test of the CWA functions. You might want to check
> the existing blog entry
> (https://www.coronawarn.app/de/blog/2020-12-28-corona-warn-app-version-1-10/)
> for the intention of the contact diary feature. So, if your intention
> is *not* to share certain contacts with the health authority, this is
> your responsibility. But obviously, there is no need to share your
> personal list of documented meetings with all of you contacts - again,
> this is your data and you select carefully what you share and who
> receives these information.

Ok, I see that my wording was a bit misleading, my apologies. 
To make it clear: I do not accuse you or the team of deliberately
violating the principles of data protection 

I do understand that data protection is a very important design
decision, even if it means not to allow a GPS functionality optionally
since it would violate the Google Apple ENA implementation guidelines.

What I wanted to point out were fats motivated by a purely practical
point of view.

The health authorities in Germany are currently overloaded with work.
Heck, they sometimes have to rely on Fax because of the lack of software
compatibility.

Sending them a complete list of your contacts is a fine thing, but it
ignores a bit the reality here.

That is why informing your contacts individually is even more important.

Now from a theoretical point of view you are right, the app does not
prevent you from sending this information. 

If you posses an iPhone (12) max, it might be even doable to cut, copy and
past single contacts and send them, but most of the users are forced to
send this list to their laptops/desktops and process the list there. 

That is not precisely user friendly or encouraging. 

Mind you that still a lot of people do not upload their test results
into the app, one reason could be that they are not encouraged or
reminded enough.

In a nutshell, it would be very very useful to make sending this
information individually as simple as possible, not just simply possible.

I hope I explained my point better.

Kind regards

Uwe Brauer
